conjure-java-core/src/main/java/com/palantir/conjure/java/types/ReturnTypeClassNameVisitor.java

```diff
@@ -58,6 +59,10 @@ public TypeName visitMap(MapType type) {
 
     @Override
     public TypeName visitOptional(OptionalType type) {
+        if (type.getItemType().accept(TypeVisitor.IS_PRIMITIVE)
+                && type.getItemType().accept(TypeVisitor.PRIMITIVE).equals(PrimitiveType.BINARY)) {
+            return delegate.visitPrimitive(PrimitiveType.BINARY);
```

Comment:
I think we want this.visit, not delegate, since this implementation provides return-type specific binary classes.

Reply:
fixed